src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunctionImpl.cs

```diff
@@ -256,8 +298,14 @@ public void Visit(RecordValue recordValue)
                 {
                     _writer.WriteStartObject();
 
-                    foreach (NamedValue namedValue in recordValue.Fields)
+                    foreach (NamedValue namedValue in recordValue.Fields.OrderBy(f => f.Name, StringComparer.Ordinal))
                     {
+                        if (_checkCancellation(0))
+                        {
+                            _writer.WriteEndObject();
```

Comment:
>_writer.WriteEndObject() [](http://example.com/codeflow?start=28&length=24)

Do we need to write the end object, since it will be ignored (as there is an error)?

Reply:
agreed. Check should just throw, and so write end is a non-issue

Reply:
When reaching the max depth, we were returning a valid string.
I'll make it throw